Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.Serialization;
+
+namespace Mirror
+{
+    [AddComponentMenu("Network/NetworkLobbyManager")]
+    [HelpURL("https://vis2k.github.io/Mirror/Components/NetworkLobbyManager")]
+    public class NetworkLobbyManager : NetworkManager
+    {
+        struct PendingPlayer
+        {
+            public NetworkConnection conn;
+            public GameObject lobbyPlayer;
+        }
+
+        // configuration
+        [Header("Lobby Settings")]
+        [FormerlySerializedAs("m_ShowLobbyGUI")] [SerializeField] internal bool showLobbyGUI = true;
+        [FormerlySerializedAs("m_MinPlayers")] [SerializeField] int minPlayers = 1;
+        [FormerlySerializedAs("m_LobbyPlayerPrefab")] [SerializeField] NetworkLobbyPlayer lobbyPlayerPrefab;
+
+        [Scene]
+        public string LobbyScene;
+
+        [Scene]
+        public string GameplayScene;
+
+        // runtime data
+        [FormerlySerializedAs("m_PendingPlayers")] List<PendingPlayer> pendingPlayers = new List<PendingPlayer>();
+        List<NetworkLobbyPlayer> lobbySlots = new List<NetworkLobbyPlayer>();
+
+        internal bool allPlayersReady = false;
```

Comment:
why internal?
and can't this be a property where get{} checks if all players are ready? then we wouldn't have to set this to true/false manually.

Reply:
This is set false based on specific events, there's no reason to re-evaluate all players in OnGUI.